docs/how_to_connector.rst

```diff
@@ -0,0 +1,146 @@
+How to create a connector
+-------------------------
+
+On pykiso view, a connector is the communication medium between the
+auxiliary and the device under test. It can consist of two different blocks:
+
+- a **communication channel**
+- a **flasher**
+
+Thus, its goal is to implement the
+sending and reception of data on the lower level protocol layers
+(e.g. CAN, UART, UDP).
+
+Both can be used as context managers as they inherit the common interface
+:py:class:`~pykiso.connector.Connector`.
+
+.. note::
+  Many already implemented connectors are available under :py:mod:`pykiso.lib.connectors`.
+  and can easily be adapted for new implementations.
+
+Communication Channel
+~~~~~~~~~~~~~~~~~~~~~
+
+In order to facilitate the implementation of a communication channel and to ensure
+the compatibility with different auxiliaries, pykiso provides a common
+interface :py:class:`~pykiso.connector.CChannel`.
+
+This interface enforces the implementation of the following methods:
+
+- :py:meth:`~pykiso.connector.CChannel._cc_open`: open the communication.
+  Does not take any argument.
+- :py:meth:`~pykiso.connector.CChannel._cc_close`: close the communication.
+  Does not take any argument.
+- :py:meth:`~pykiso.connector.CChannel._cc_send`: send data if the communication is open.
+  Requires one positional argument ``msg`` and one keyword argument ``raw``, used to serialize the data
+  before sending it.
+- :py:meth:`~pykiso.connector.CChannel._cc_receive`: receive data if the communication is open.
+  Requires one positional argument ``timeout`` and one keyword argument ``raw``, used to deserialize
+  the data when receiving it.
+
+
+Class definition and instanciation
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To create a new communication channel, the first step is to define its class
+and constructor.
+
+Let's admin that the following code is added to a file called **my_connector.py**:
+
+.. code:: python
+
+    import pykiso
+
+    MyCommunicationChannel(pykiso.CChannel):
+
+        def __init__(self, arg1, arg2, kwarg1 = "default"):
+            ...
+
+Then, if this CChannel has to be used within a test, the test configuration file
+will derive from its location and constructor parameters:
+
+.. code:: yaml
+
+    connectors:
+      my_chan:
+        # provide the constructor parameters
+        config:
+          # arg1 and arg2 are mandatory as we defined them as positional arguments
+          arg1: value for positional argument arg1

```

Comment:
I may have read fast, but did we describe somewhere that the arguments of __init__() are linked to the arguments passed here?

Reply:
Yes, the lines above show the constructor, this yaml sample is there to illustrate the config file structure resulting of such a constructor